pkg/platform/authentication/auth.go

```diff
@@ -113,9 +116,40 @@ func (s *Auth) Close() error {
 	return nil
 }
 
+func (s *Auth) storeAuthenticatedDevice(deviceCode strfmt.UUID, response *oauth.AuthDeviceGetOK) error {
+	defer s.updateRollbarPerson()
+	s.user = response.Payload.AccessToken.User
+	s.bearerToken = response.Payload.AccessToken.Token
+	clientAuth := httptransport.BearerToken(s.bearerToken)
+	s.clientAuth = &clientAuth
+	err := s.cfg.Set(deviceCodeConfigKey, deviceCode)
+	if err != nil {
+		return errs.Wrap(err, "Could not set deviceCode credentials in config")
+	}
+	return err
+}
+
 // Authenticated checks whether we are currently authenticated
 func (s *Auth) Authenticated() bool {
-	return s.clientAuth != nil
+	if s.clientAuth != nil {
+		return true
+	} else if existingDeviceCode := s.cfg.GetString(deviceCodeConfigKey); existingDeviceCode != "" && s.bearerToken == "" {
+		// Check if the device is still authenticated with the Platform and if so, get a token.
+		deviceCode := strfmt.UUID(existingDeviceCode)
+		params := oauth.NewAuthDeviceGetParams()
+		params.SetDeviceCode(deviceCode)
+		if response, err := mono.Get().Oauth.AuthDeviceGet(params); response != nil {
```

Comment:
The `err` here is not being checked here, just that the `response` is not `nil`

Reply:
I've looked at the generated swagger code and if response is `nil`, then `err` is not nil and vice-versa. I could change it to be an `} else if err != nil {`. Is that preferrable?

Reply:
Response being nil seems like an error to me, so checking the error feels like the correct thing to do here unless there's a legitimate reason for there not being a response?

Reply:
This code has been changed. I cannot find this snippet anywhere else, so I think this has been addressed.

The reason I wasn't worried about errors is that response and err are mutually exclusive according to swagger. If response is not nil, there won't be an error, and vice-versa.